sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -292,7 +293,11 @@ trait PrunedFilteredScan {
  */
 @Stable
 trait InsertableRelation {
-  def insert(data: DataFrame, overwrite: Boolean): Unit
+  def insert(data: DataFrame, overwrite: Boolean): Unit = {}
+
+  def insert(data: DataFrame, overwrite: Boolean, metrics: Map[String, SQLMetric]): Unit = {
```

Comment:
The source isn't necessarily related to files. It's a bit odd to assume the number of files here. Also, seems like this is what DSv2 should cover?

Reply:
Does DSv2 have the similar interface? Current Delta Lake is using this command and besides this metrics, we have implemented ensure distribution for InsertIntoDataSource to support bucketing delta table. Does DSv1 accepte no more improvements? It has to move to DSv2?